### ISSUE TITLE
fortiswitch template port names as $1

### ISSUE THREAD

**Issue by ftoledo** [zabbix/community-templates#300]
I triying to use the fortiswitch_124e-poe.yaml template on F148F POE
 
![imagen](https://github.com/zabbix/community-templates/assets/95441/61b6a9da-1ec5-444e-a734-4f854f3f4b03)
the port names sems to wont detected as show as $1

any way to fix ?
thanks!

**Comment by fxbeaulieu** [comment]
I have this issue since migrating to ZBX 6.0 in various templates too. If anyone has a solution or even an explanation as to why this happens, I would love to hear it.

**Comment by tbblake** [comment]
Positional macros were deprecated in 4.0 and fully removed in 6.0.

https://www.zabbix.com/documentation/6.0/en/manual/introduction/whatsnew600#positional-macros-no-longer-supported

**Comment by ftoledo** [comment]
> I have this issue since migrating to ZBX 6.0 in various templates too. If anyone has a solution or even an explanation as to why this happens, I would love to hear it.

I found that other templates change the variables $1 $2 $nnnn by names
example:
https://github.com/zabbix/community-templates/pull/286/files

**Comment by tbblake** [comment]
That template used too, that pull request changes the template from using positional macros to using discovery macros.

**Comment by ftoledo** [comment]
I still don't have knowledge of template development. (I just now start to reading docs)

for this case:
![imagen](https://github.com/zabbix/community-templates/assets/95441/05ccb594-c62e-41c6-ae22-800ec423e9f8)

Should I change the $1 to {#SNMPVALUE} ?

**Comment by tbblake** [comment]
I believe so.  Personally, I would import the templates and look at the item prototype.  You'd likely see an item prototype with a name of Link Speed $1 and a key of [ifHighSpeed[{#SNMPVALUE}].  I'd change the $1 to {#SNMPVALUE} like you say, re-run discovery or wait for it to run, and see if it updates for the Link Speed items.  If it does, then make similar changes on the other item prototypes.  If they all work, I'd then export the template and do a pull request.

**Comment by tbblake** [comment]
Yup looks like just these three item prototypes would need that change.

![image](https://github.com/zabbix/community-templates/assets/45319489/5cb7adb1-0afd-40df-979a-85ee675ca863)

**Comment by ftoledo** [comment]
Yes! i can conrfim that change $1 to {#SNMPVALUE} just works.

![imagen](https://github.com/zabbix/community-templates/assets/95441/ba59c884-c58f-48cc-becc-c3afe9632284)

**Comment by fxbeaulieu** [comment]
Thanks a lot, will try to correct our templates in the same way tomorrow
